pulsar-metadata/src/main/java/org/apache/pulsar/metadata/cache/impl/MetadataCacheImpl.java

```diff
@@ -58,6 +61,7 @@ public class MetadataCacheImpl<T> implements MetadataCache<T>, Consumer<Notifica
     private final MetadataStore store;
     private final MetadataStoreExtended storeExtended;
     private final MetadataSerde<T> serde;
+    private final ScheduledExecutorService backoffExecutor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
It' better not use `Executors....` to new an Executor.
BTW, can we reuse the existing scheduler?  Such as `AbstractMetadataStore.executor`, maybe we can increase it's thread num to more.

Reply:
But how can we access the `AbstractMetadataStore.executor` here?

Reply:
can we just pass `AbstractMetadataStore.executor` to  MetaCacheImp via the constructors?

Reply:
> It' better not use Executors.... to new an Executor.

Why?

Reply:
Because it's thread name is meaningless, it maybe could lead to some difficulties when we analysis the thread dump.

Reply:
I get your point. Then we can use `Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("metadata-cache-backoff"))` to give it a meaningful name. https://github.com/apache/pulsar/pull/23686/commits/dbca96ebb98fc682131c779077e5d392e94a0352

WDYT?